### PR TITLE
makefile add license check support

### LINKED ISSUE #39: Tooling should validate copyright header is present

### PR DESCRIPTION
**Pull request by WPH95**
resolved #39

### CODE REVIEW

**Review by djaglowski — COMMENTED** [review]
When I run `make ci-check`, I get some errors on generated files:

```
make ci-check
GOOS=darwin /Library/Developer/CommandLineTools/usr/bin/make for-all CMD="go vet ./..."
GOOS=linux /Library/Developer/CommandLineTools/usr/bin/make for-all CMD="go vet ./..."
GOOS=windows /Library/Developer/CommandLineTools/usr/bin/make for-all CMD="go vet ./..."
golangci-lint run ./...
addlicense FAILED => add License errors:

./testutil/operator_builder.go
./testutil/pipeline.go
./testutil/database.go
./testutil/operator.go

Use 'make add-license' to fix this.
make: *** [check-license] Error 1
```

@tigrannajaryan Is it an acceptable approach to exclude these files from the license check?

**Comment by WPH95** [comment]
if want ignore testutil/*
just change makefile `ALL_SRC`  like https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/Makefile.Common#L5

**Comment by djaglowski** [comment]
I think that this PR should include making all the license checks pass, so we should include the change here. Do you mind adding?

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,8 @@ go 1.14
 require (
 	github.com/antonmedv/expr v1.8.2
 	github.com/cenkalti/backoff/v4 v4.0.2
+	github.com/golangci/golangci-lint v1.37.1 // indirect
+	github.com/google/addlicense v0.0.0-20200906110928-a0294312aa76 // indirect
```
Comment (djaglowski) [inline go.mod:9]: I don't think we need to include these in the module, and believe we should try not to.

Neither the collector or collector-contrib repos have needed these in `go.mod`.

File: go.sum [hunk 2]
```diff
@@ -1,144 +1,666 @@
+4d63.com/gochecknoglobals v0.0.0-20201008074935-acfc0b28355a h1:wFEQiK85fRsEVF0CRrPAos5LoAryUsIX1kPW/WrIqFw=
```
Comment (djaglowski) [inline go.sum:1]: This is a lot of extras that I can't reconcile. Can you try regenerating this after addressing the `go.mod` comment?

**Review by djaglowski — CHANGES_REQUESTED** [review]
We need to figure out how to make use of the tooling packages without including them in the module file.

**Comment by tigrannajaryan** [comment]
> When I run `make ci-check`, I get some errors on generated files:
> 
> ```
> make ci-check
> GOOS=darwin /Library/Developer/CommandLineTools/usr/bin/make for-all CMD="go vet ./..."
> GOOS=linux /Library/Developer/CommandLineTools/usr/bin/make for-all CMD="go vet ./..."
> GOOS=windows /Library/Developer/CommandLineTools/usr/bin/make for-all CMD="go vet ./..."
> golangci-lint run ./...
> addlicense FAILED => add License errors:
> 
> ./testutil/operator_builder.go
> ./testutil/pipeline.go
> ./testutil/database.go
> ./testutil/operator.go
> 
> Use 'make add-license' to fix this.
> make: *** [check-license] Error 1
> ```
> 
> @tigrannajaryan Is it an acceptable approach to exclude these files from the license check?

@djaglowski Yes, we normally exclude generated files from license checks.

**Pull request closed without merging**